Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -440,8 +440,21 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
         return AccessibilityRole::LandmarkMain;
     if (node()->hasTagName(navTag))
         return AccessibilityRole::LandmarkNavigation;
-    if (node()->hasTagName(asideTag))
+    // The aside element should not assume the complementary role when nested
+    // within the sectioning content elements.
+    // https://w3c.github.io/html-aam/#el-aside
```

Comment:
This is a good comment — let's move it closer to the code that it describes (`if (isDescendantOfElementType({ asideTag, articleTag, sectionTag, navTag }))`)

Reply:
Sure!